main/main.c

```diff
@@ -96,7 +96,10 @@ signal_handler(int signum)
 		fprintf(stderr, "caught SIGINT\n");
 	else if (signum == SIGTERM)
 		fprintf(stderr, "caught SIGTERM\n");
-	else
+	else if (signum == SIGPIPE) {
+		fprintf(stdout, "caught SIGPIPE\n");
+		return;
+	} else
 		fprintf(stderr, "caught unknown signal\n");
```

Comment:
Add `signum` to the error message. It'll be helpful if it's ever triggered in production.